arcgis-ios-sdk-samples/Geometry/Geodesic operations/GeodesicOperationsViewController.swift

```diff
@@ -0,0 +1,105 @@
+        
+        // Initialize map with imagery basemap.
+        let map = AGSMap(basemap: AGSBasemap.imagery())
+        
+        // Assign map to map view.
+        mapView.map = map
+        
+        // Add graphics overlay to the map view.
+        mapView.graphicsOverlays.add(self.graphicsOverlay)
+        
+        // Set map view touch delegate to self.
+        mapView.touchDelegate = self
+        
+        // Add origin graphic, destination graphic, and path graphic.
+        addGraphics()
+    }
+    
+    func addGraphics() {
+        // Create graphic symbols.
+        let locationMarker = AGSSimpleMarkerSymbol(style: .cross, color: .blue, size: 10)
+        let pathSymbol = AGSSimpleLineSymbol(style: .dash, color: .blue, width: 5)
+        
+        // Create origin graphic.
+        let originGraphic = AGSGraphic(geometry: JFKAirportLocation, symbol: locationMarker, attributes: nil)
+        graphicsOverlay.graphics.add(originGraphic)
+        
+        // Create destination graphic.
+        destinationGraphic = AGSGraphic(geometry: nil, symbol: locationMarker, attributes: nil)
+        graphicsOverlay.graphics.add(destinationGraphic)
+    
+        // Create graphic to represent geodesic path between origin and destination.
+        pathGraphic = AGSGraphic(geometry: nil, symbol: pathSymbol, attributes: nil)
+        graphicsOverlay.graphics.add(pathGraphic)
+    }
+    
+    // MARK: - AGSGeoViewTouchDelegate
+    
+    func geoView(_ geoView: AGSGeoView, didTapAtScreenPoint screenPoint: CGPoint, mapPoint: AGSPoint) {
+        // If callout is open, dismiss it.
+        if !mapView.callout.isHidden {
+            mapView.callout.dismiss()
+        }
+        
+        // Get the tapped geometry projected to WGS84.
+        guard let destinationLocation = AGSGeometryEngine.projectGeometry(mapPoint, to: .wgs84()) as? AGSPoint else {
+            return
+        }
+        
+        // Update geometry of destination graphic.
+        destinationGraphic.geometry = destinationLocation
+        
+        // Create a straight line path from origin to destination.
+        let path = AGSPolyline(points: [JFKAirportLocation, destinationLocation])
+        
+        // Densify the polyline to show the geodesic curve.
+        guard let geodeticPath = AGSGeometryEngine.geodeticDensifyGeometry(path, maxSegmentLength: 1, lengthUnit: .kilometers(), curveType: .geodesic) else {
+            return
+        }
+        
+        // Update geometry of path graphic.
+        pathGraphic.geometry = geodeticPath
+        
+        // Calculate geodetic distance between origin and destination.
+        let distance = AGSGeometryEngine.geodeticLength(of: geodeticPath, lengthUnit: .kilometers(), curveType: .geodesic)
+        
+        // Display the distance in mapview's callout.
+        mapView.callout.title = "Distance"
+        mapView.callout.detail = String(format: "%.2f km", locale: Locale.current, distance)
```

Comment:
You might consider using a `MeasurementFormatter` so that the measurement is formatted properly for the user's locale.

To do so, add the formatter as a property of the view controller. You do not need to set the locale as the default is `current`:

```swift
let measurementFormatter = MeasurementFormatter()
```

Then construct a measurement and use the formatter to create the callout detail string:

```swift
let measurement = Measurement<UnitLength>(value: distance, unit: .kilometers)
mapView.callout.detail = measurementFormatter.string(from: measurement)
```

Reply:
Changed that.